core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulseDataFetcher.java

```diff
@@ -114,11 +123,12 @@ public void prepareQuickPulseDataForSend() {
     }
 
     @VisibleForTesting
-    String getEndpointUrl() {
-        return getQuickPulseEndpoint() + "/post?ikey=" + getInstrumentationKey();
+    String getEndpointUrl(String endpointURL) {
+        return endpointURL + "/post?ikey=" + getInstrumentationKey();
```

Comment:
or maybe `endpointPrefix`?
```suggestion
    String getEndpointUrl(String endpoint) {
        return endpoint + "/post?ikey=" + getInstrumentationKey();
```